testclock/clock_test.go

```diff
@@ -160,6 +226,28 @@ func (*clockSuite) TestNewTimerReset(c *gc.C) {
 	}
 }
 
+func (*clockSuite) TestNewAlarmReset(c *gc.C) {
+	t0 := time.Now()
+	cl := testclock.NewClock(t0)
+	alarm := cl.NewAlarm(cl.Now().Add(time.Second))
+	cl.Advance(time.Second)
+	select {
+	case t := <-alarm.Chan():
+		c.Assert(t.UTC(), gc.Equals, t0.Add(time.Second).UTC())
+	case <-time.After(longWait):
+		c.Fatalf("expected event to be triggered")
+	}
+
+	alarm.Reset(cl.Now().Add(50 * time.Millisecond))
```

Comment:
Should be testing the return result, both for true and false cases

Reply:
Good catch.